lib/ansible/module_utils/postgres.py

```diff
@@ -59,3 +61,30 @@ def postgres_common_argument_spec():
         ssl_mode=dict(default='prefer', choices=['allow', 'disable', 'prefer', 'require', 'verify-ca', 'verify-full']),
         ca_cert=dict(aliases=['ssl_rootcert']),
     )
+
+
+def connect_to_db(module, kw, autocommit=False):
```

Comment:
Would it make sense to move build the `kw`s inside this function? I think we should be able to move the following section into here since we already have the module so it should be all we need to build the connection details.
```python
params_map = {
        "login_host": "host",
        "login_user": "user",
        "login_password": "password",
        "port": "port",
        "db": "database",
        "ssl_mode": "sslmode",
        "ca_cert": "sslrootcert"
    }
    kw = dict((params_map[k], v) for (k, v) in iteritems(module.params)
              if k in params_map and v != "" and v is not None)
    # If a login_unix_socket is specified, incorporate it here.
    is_localhost = "host" not in kw or kw["host"] == "" or kw["host"] == "localhost"
    if is_localhost and module.params["login_unix_socket"] != "":
        kw["host"] = module.params["login_unix_socket"]
    if psycopg2.__version__ < '2.4.3' and sslrootcert is not None:
        module.fail_json(msg='psycopg2 must be at least 2.4.3 in order to user the ca_cert parameter')
```
It would reduce the duplicate code and might make it easier to manage in the future, since every module is now using the `postgres_common_argument_spec` they should have the correct params too.

Reply:
@andytom , very good note, thank you! I should think and try

Reply:
@andytom , I think it's necessary to keep kw because we return kw that contains connection params as well.

Reply:
I suggest to merge this PR and after that we can think about optimizations.

Reply:
@andytom , I made a short note in [#PostgreSQL](https://github.com/ansible/community/wiki/PostgreSQL).
The first aim of this PR is to move connection_to_db and get_pg_version to module_utils (as gundalow and I wanted). However your suggestion is very interesting, so after merge this one we should think how to implement it safely

Reply:
@andytom Good suggestion thanks. Let's get this merged and backported then we can look at further improvements.